### DISCUSSION TITLE
Nextjs App deployed on Vercel , only first opened page has problem with sending request to db

### DISCUSSION THREAD

**Discussion by n1kalai** [vercel/next.js#17525] [Help] [answered]
hello guys, i deployed nextjs app on vercel, i can visit the website, but first page, no matter which url, gives me 500 error, then i change between pages and same page which gave me 500, gives 200 and shows data, i use getInitialProps, tried getServerSideProps but no result ... can anybody help ? :(

**Comment by ErnestBrandi** [comment]
Can you give the link to that website ? And ideally the code of the homepage (at least the server side part).

**Reply by n1kalai** [reply]
i have db on localhost and link couldnot help you :( 
i have tried this:
MainPage.getInitialProps = async (context) => {
try {
  
    const sales = await baseURL.get("/sales/getsales");
    const navbar = await baseURL.get("/category/getcategories");
    if(sales && navbar){
    return { sales: sales.data, navbar: navbar.data };
    }
return {sales:[],navbar:[]}
} catch (err) {
  console.log(err)
}
}
and this without try catch, don't know what goes on :|
MainPage.getInitialProps = async (context) => {

    const sales = await baseURL.get("/sales/getsales");
    const navbar = await baseURL.get("/category/getcategories");
    if(sales && navbar){
    return { sales: sales.data, navbar: navbar.data };
    }
return {sales:[],navbar:[]}
};

**Comment by n1kalai** [comment, marked as answer]
i tried it on real domain db and worked...
